keps/sig-node/20191014-configurable-system-reserved.md

```diff
@@ -0,0 +1,183 @@
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+Kubelet supports configuring a subset of the Node resources it shall not actively manage via the "system-reserved" flag.
+Currently only CPU, and memory are supported. 
+The CPU Manager -more precisely its static policy- implemented within Kubelet interprets the absolute amount of "CPU shares" coming from this configuration as the "first X number of core IDs I shall not touch".
+X denotes the smallest amount of vCPUs (threads, or physical cores) which are enough to satisfy the request.
+I.e. for system-reserved=1200m CPU Manager excludes CPU cores 0, and 1 form the list of CPU cores usable by K8s workloads.
+This KEP proposes to enhance the format, and the meaning of this flag so that it becomes possible to configure an exact list of CPU cores Kubelet shall entirely exclude from its supervision, under all circumstances.
+The exclusion should be always applicable independently from other CPU management related configuration options such as enablement of CPU Manager, or the Node's CPU management policy (i.e. "none", or "static").
+
+## Motivation
+Kubelet always assumes that it is the primary software component managing the CPU cores of the host.
+However, in certain infrastructures this might not always be the case.
+While it is already possible to effectively take-away CPU cores from the Kubernetes managed workloads via the kube-reserved and system-reserved kubelet flags, the current implementation has two perceived shortcomings.
+1: reservations are tied to the CPU Manager, and its static policy
+2: this implicit way of declaring a Kubernetes managed CPU pool is not flexible enough to cover all use-cases
+
+To accommodate the use-cases -mainly related to seamlessly inter-working with multiple resource managers on the same node-, the need arises to enhance existing CPU management implementation to be able to explicitly, and universally define a discontinuous pool of non-manageable CPUs.
+Such feature could come in handy if one would like to:
+- ensure proper resource accounting and separation between systemd processes and Kubernetes managed workloads
+- ensure proper resource accounting and separation within a hybrid infrastructure (e.g. Openstack + Kubernetes resource managers running on the same node etc.)
```

Comment:
simple question is that a openstack-helm or kubernetes-kuryr or both of them or another?

Reply:
I tried to refer to the scenario where Openstack and K8s simply co-exist on the same node - without any of them managing the other.
So K8s is not running in VMs, and Openstack is not running in Pods.
But even if OS is running in Pods, the situation is possibly the same as I guess Nova container has access to the full cgroup hierarchy of the underlying host